### ISSUE TITLE
使用export_model.py导出PicoDet轻量级主体检测模型，其后处理部分与官方提供的inference 模型不一样，Head部分也不一样

### ISSUE THREAD

**Issue by edition3234** [PaddlePaddle/PaddleDetection#8249]
### 问题确认 Search before asking

- [X] 我已经查询[历史issue](https://github.com/PaddlePaddle/PaddleDetection/issues)，没有发现相似的bug。I have searched the [issues](https://github.com/PaddlePaddle/PaddleDetection/issues) and found no similar bug report.

### Bug描述 Describe the Bug

https://github.com/PaddlePaddle/PaddleClas/blob/release/2.5/docs/zh_CN/training/PP-ShiTu/mainbody_detection.md

在这个网址下载的预训练模型和inference 模型，尝试过使用paddleDect2.3和2.6两个版本的export_model.py，导出预训练模型，开启后处理和nms，得到的网络后处理部分和inference 模型不一样。

inference 模型里有像  while  tensor_array_to_tensor  这样的自定义算子，在paddleDect工程中找不到这种算子的定义，只找到一个multiclass_nms。

导出模型中Head部分与inference 模型的Head部分也不一样，使用的是官方提供的picodet_lcnet_x2_5_640_mainbody.yml文件，head选择的是PicoHead网络。

请问inference 模型的网络定义，是在哪个工程里？

picodet_lcnet_x2_5_640_mainbody.yml内容：
`_BASE_: [
  '/configs/runtime.yml',
  '/configs/picodet/legacy_model/_base_/picodet_esnet.yml',
  '/configs/picodet/legacy_model/_base_/optimizer_100e.yml',
  '/configs/picodet/legacy_model/_base_/picodet_640_reader.yml',
]

pretrain_weights: https://paddledet.bj.bcebos.com/models/pretrained/LCNet_x2_5_ssld_pretrained.pdparams
weights: output/picodet_lcnet_x2_5_640_mainbody/model_final
find_unused_parameters: True
use_ema: true
cycle_epoch: 20
snapshot_epoch: 2

PicoDet:
  backbone: LCNet
  neck: CSPPAN
  head: PicoHead

LCNet:
  scale: 2.5
  feature_maps: [3, 4, 5]

metric: COCO
num_classes: 1

TrainDataset:
  !COCODataSet
    image_dir: ./
    anno_path: train.json
    dataset_dir: dataset/mainbody/
    data_fields: ['image', 'gt_bbox', 'gt_class', 'is_crowd']

EvalDataset:
  !COCODataSet
    image_dir: ./
    anno_path: val.json
    dataset_dir: dataset/mainbody/

TestDataset:
  !ImageFolder
    anno_path: ./dataset/mainbody/val.json`

### 复现环境 Environment

- OS : linux
- python : 3.8.10

1.
- paddlepaddle : 2.4.2
- paddleDetection : 2.6

2.
- paddlepaddle : 2.1.2
- paddleDetection : 2.3

### Bug描述确认 Bug description confirmation

- [X] 我确认已经提供了Bug复现步骤、代码改动说明、以及环境信息，确认问题是可以复现的。I confirm that the bug replication steps, code change instructions, and environment information have been provided, and the problem can be reproduced.

### 是否愿意提交PR？ Are you willing to submit a PR?

- [ ] 我愿意提交PR！I'd like to help by submitting a PR!

**Comment by Andy197527** [comment]
解决了吗？
